apps/fabric-website/src/components/Site/Site.tsx

```diff
@@ -126,13 +114,18 @@ export class Site<TPlatforms extends string = string> extends React.Component<IS
     if (prevState.pagePath !== this.state.pagePath) {
       this._jumpToAnchor(extractAnchorLink(location.hash));
     }
-  }
 
-  public render() {
-    if (this.state.initialRender) {
-      return null;
+    const { activePages, pagePlatforms } = this.state;
+    const { siteDefinition } = this.props;
+
+    // If current page doesn't have pages for the active platform, switch to its first platform.
+    if (Object.keys(pagePlatforms).length > 0 && activePages.length === 0) {
```

Comment:
Thanks for fixing this up! One comment, is there any way we can do this part in the constructor too? Otherwise this might end up causing a flash on first render.

Reply:
I can do it in both.

Reply:
I'm a bit confused--if we're doing this in the constructor, why do we still need it in componentDidUpdate? Is that to handle when switching to a different area?

Reply:
Yeah, if for some reason the user has a state that includes a platform that doesn't exist/have any pages, this will update it to the first platform defined, like if we rename/remove a platform. It needs to be in both for the reason you just said.